kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/Console.java

```diff
@@ -372,80 +391,82 @@ private HttpContext initResourceContext(final HttpContext defaultContext) {
     private synchronized void initHTTPService() throws NamespaceException, ServletException {
 
         final HttpContext defaultContext = this.httpService.createDefaultHttpContext();
-        final HttpContext resourceContext = initResourceContext(defaultContext);
-        sessionContext = initSessionContext(defaultContext);
 
-        this.httpService.registerResources(ADMIN_ROOT, "www", resourceContext);
-        this.httpService.registerResources(AUTH_PATH, "www/auth.html", sessionContext);
-        this.httpService.registerResources(CONSOLE_PATH, "www/denali.html", sessionContext);
+        this.resourceContext = initResourceContext(defaultContext);
+        this.sessionContext = initSessionContext(defaultContext);
+
+        this.httpService.registerResources(ADMIN_ROOT, "www", this.resourceContext);
+        this.httpService.registerResources(AUTH_PATH, "www/auth.html", this.sessionContext);
+        this.httpService.registerResources(this.appRoot, "www/denali.html", this.sessionContext);
         this.httpService.registerServlet(LOGIN_MODULE_PATH + "/banner", new GwtBannerServiceImpl(), null,
-                resourceContext);
+                this.resourceContext);
 
-        this.httpService.registerServlet("/", new RedirectServlet("/"::equals, this.appRoot), null, resourceContext);
-        this.httpService.registerServlet(AUTH_RESOURCE_PATH, new SendStatusServlet(404), null, resourceContext);
-        this.httpService.registerServlet(CONSOLE_RESOURCE_PATH, new SendStatusServlet(404), null, resourceContext);
+        this.httpService.registerServlet("/", new RedirectServlet("/"::equals, () -> this.appRoot), null,
+                this.resourceContext);
+        this.httpService.registerServlet(AUTH_RESOURCE_PATH, new SendStatusServlet(404), null, this.resourceContext);
+        this.httpService.registerServlet(CONSOLE_RESOURCE_PATH, new SendStatusServlet(404), null, this.resourceContext);
 
         this.httpService.registerServlet(PASSWORD_AUTH_PATH,
-                new GwtPasswordAuthenticationServiceImpl(this.authMgr, CONSOLE_PATH), null, sessionContext);
+                new GwtPasswordAuthenticationServiceImpl(this.authMgr, () -> this.appRoot), null, this.sessionContext);
```

Comment:
The `appRoot` parameter was originally meant to allow having a 3rd party web ui available on the root. It is not related the console path. Maybe we can improve the parameter description to avoid misunderstanding. If you want to enable the customisation of the console path (which can be desirable) please introduce another configuration parameter.

Reply:
I don't understand what means this appRoot is. If change the appRoot this code
`
this.httpService.registerServlet("/", new RedirectServlet("/"::equals, this.appRoot), null, resourceContext);
`
will redirect console to the changed appRoot. That's why i thought appRoot is console path.

By the way i don't know why Eclipse CLA check is aways failing.

Reply:
Should i change to this?
`this.httpService.registerServlet("/", new RedirectServlet("/"::equals, CONSOLE_PATH), null, resourceContext);`

If not next time the we can't enter console by root path '/'.

Reply:
I will try to explain better the use case for `appRoot`:

Suppose you want to add another web application to the gateway that is completely different from Kura web ui, for example a web application developed by a customer that manages only its application, and the customer wants it to be available on the root path `/`.

You can achieve this in the following way:

1. Create a custom bundle that serves the application on `/mycompany/myapp`.
2. Set the `appRoot` parameter to `/mycompany/myapp`.

In this way the custom application will be available on the root path, Kura console will be still available on `/admin/console` which is currently hardcoded.

In other words `appRoot` is simply the relative path to which the user will be redirected when accessing the root path, which is not necessarily the Kura console path.

Reply:
Thanks. I see.  You can review the change of #2568 about this and i'll close this PR and open other PR later because of Eclipse CLA failing